core/templates/domain/state/StateObjectFactory.ts

```diff
@@ -109,6 +111,40 @@ export class State {
     this.writtenTranslations = otherState.writtenTranslations;
     this.nextContentIdIndex = otherState.nextContentIdIndex;
   }
+
+  getRequiredWrittenTranslationsCount(): number {
+    let interactionId = this.interaction.id;
+    const contentIdsToRemove = [];
+
+    let allContentIds = this.writtenTranslations.getAllContentId();
+
+    // As of now we do not delete interaction.hints when a user deletes
+    // interaction, so these hints audio are not counted in checking
+    // status of a state.
+    if (!interactionId ||
+      INTERACTION_SPECS[interactionId].is_linear ||
+      INTERACTION_SPECS[interactionId].is_terminal) {
+      const hintContentIds = allContentIds.filter(
+        contentId => contentId.indexOf(AppConstants.COMPONENT_NAME_HINT) === 0);
+      contentIdsToRemove.push(...hintContentIds);
+      // Excluding default_outcome content status as default outcome's
+      // content is left empty so the translation or voiceover is not
+      // required.
+      contentIdsToRemove.push('default_outcome');
+    }
+
+    // As of now, there are no ways of contributing rule input
```

Comment:
Add TODO to remove this, pointing to the same issue that was mentioned before (see review comments in previous files).